src/plugins/workspace/server/plugin.ts

```diff
@@ -64,12 +70,63 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
     });
   }
 
+  private setupPermission(core: CoreSetup, { applicationConfig }: AppPluginSetupDependencies) {
+    this.permissionControl = new SavedObjectsPermissionControl(this.logger);
+
+    core.http.registerOnPostAuth(async (request, response, toolkit) => {
+      let groups: string[];
+      let users: string[];
+      let configGroups: string[];
+      let configUsers: string[];
+
+      // There may be calls to saved objects client before user get authenticated, need to add a try catch here as `getPrincipalsFromRequest` will throw error when user is not authenticated.
+      try {
+        ({ groups = [], users = [] } = this.permissionControl!.getPrincipalsFromRequest(request));
+      } catch (e) {
+        return toolkit.next();
+      }
+
+      if (!!applicationConfig) {
+        const [coreStart] = await core.getStartServices();
+        const scopeClient = coreStart.opensearch.client.asScoped(request);
+        const applicationConfigClient = applicationConfig.getConfigurationClient(scopeClient);
+
+        const [groupsResult, usersResult] = await Promise.all([
+          applicationConfigClient
+            .getEntityConfig('opensearchDashboards.dashboardAdmin.groups')
+            .catch(() => undefined),
+          applicationConfigClient
+            .getEntityConfig('opensearchDashboards.dashboardAdmin.users')
+            .catch(() => undefined),
+        ]);
+
+        [configGroups, configUsers] = [stringToArray(groupsResult), stringToArray(usersResult)];
+      } else {
+        const globalConfig: SharedGlobalConfig = await this.globalConfig$.pipe(first()).toPromise();
+        configGroups = (globalConfig.opensearchDashboards.dashboardAdmin.groups || []) as string[];
+        configUsers = (globalConfig.opensearchDashboards.dashboardAdmin.users || []) as string[];
+      }
+      updateDashboardAdminStateForRequest(request, groups, users, configGroups, configUsers);
```

Comment:
The logic about getting OSD config from yml/application cannot be moved to utils for the time being. Because the param of `applicationConfigClient ` is not exposed, need to wait for the optimization of zilong. When the pr is prepared to OSD main, I will use the param of `applicationConfig ` to get the `applicationConfigClient`.